pkg/daemon/ceph/osd/volume.go

```diff
@@ -1007,6 +1010,29 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			Store:         "bluestore",
 		}
 
+		if !skipDeviceClass {
+			deviceProps, err := sys.GetDeviceProperties(blockPath, context.Executor)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get device %q props", blockPath)
+			}
+
+			if deviceProps["ROTA"] == "1" {
+				osd.DeviceClass = "hdd"
+			} else {
+				// if used on PVC we don't know what is device class type - ssd or nvme, setting to ssd by default
```

Comment:
On PVC we can use `RealPath` if you use `PopulateDeviceInfo()`.

Reply:
PopulateDeviceInfo() is using same GetDeviceProperties inside, so what is difference then?

Reply:
Oh nothing then, the only thing that matters is to get `RealPath` so that we can also take care of the PVC case.